From 69f946e6dfcd35eb2aab85d51e4fac13bf67e1ba Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sat, 17 Oct 2026 13:49:23 +0200
Subject: [PATCH] Fix window activation with stale event timestamps

Use 0 instead of the button event timestamp when activating a
workspace or window.

As the applet acts as a pager, it can request activation without
passing the timestamp of the button event. This avoids activation
failures with window managers that reject stale timestamps, which
can leave minimized windows in the demands-attention state instead
of restoring them.
---
 mate-window-picker-applet/task-item.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mate-window-picker-applet/task-item.c b/mate-window-picker-applet/task-item.c
index 9c6cee9..2b14b63 100644
--- a/mate-window-picker-applet/task-item.c
+++ b/mate-window-picker-applet/task-item.c
@@ -130,7 +130,7 @@ on_task_item_button_released (GtkWidget      *widget,
   {
     if (WNCK_IS_WORKSPACE (workspace) && workspace != wnck_screen_get_active_workspace (screen))
     {
-      wnck_workspace_activate (workspace, event->time);
+      wnck_workspace_activate (workspace, 0);
     }
     if (wnck_window_is_active (window))
     {
@@ -141,7 +141,7 @@ on_task_item_button_released (GtkWidget      *widget,
     }
     else
     {
-      wnck_window_activate (window, event->time);
+      wnck_window_activate (window, 0);
     }
   }
   return TRUE;
